charset/charset.go

```diff
@@ -459,6 +462,7 @@ var collations = []*Collation{
 	{247, "utf8mb4", "utf8mb4_vietnamese_ci", false},
 	{255, "utf8mb4", "utf8mb4_0900_ai_ci", false},
 	{2048, "utf8mb4", "utf8mb4_zh_pinyin_tidb_as_cs", false},
+	{2049, "utf8mb3", "utf8mb3_general_ci", true},
```

Comment:
This collation seems to be there under a different name already.

```
mysql> show collation where charset='utf8mb3' and Collation LIKE '%\_general\_ci';
+--------------------+---------+----+---------+----------+---------+---------------+
| Collation          | Charset | Id | Default | Compiled | Sortlen | Pad_attribute |
+--------------------+---------+----+---------+----------+---------+---------------+
| utf8mb3_general_ci | utf8mb3 | 33 | Yes     | Yes      |       1 | PAD SPACE     |
+--------------------+---------+----+---------+----------+---------+---------------+
1 row in set (0.01 sec)

mysql> SELECT VERSION();
+-----------+
| VERSION() |
+-----------+
| 8.0.30    |
+-----------+
1 row in set (0.01 sec)
```

There already is this:
```go
	{33, "utf8", "utf8_general_ci", false},
```

That is similar to this:
```go
	{2049, "utf8mb3", "utf8mb3_general_ci", true},
```

`utf8` is an alias for `utf8mb3`. I think the change in MySQL 8.0.28 (see https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-utf8.html ) might be related to this.

Reply:
+1, I think it's better to keep 'utf8_general_ci(33)' and just make `utf8mb3_general_ci` as an alias. 

If we want to introduce the alias, maybe we can introduce a new 'alias' table to represent the mapping relations:

* `utf8` <-> `utf8mb3`
* `utf8_bin` <-> `utf8mb3_bin`
...

Reply:
I think MySQL (and MariaDB?) are doing the mapping the other way around, so `utf8` is an alias for `utf8mb3`. The result is that tables that are created with current versions and use `utf8` show `utf8mb3` in the output. Then the plan is to switch the `utf8` alias to mean `utf8mb4` in the future. Then old tables still show `utf8mb3` while new ones show `utf8mb4` even if both were created with the same `utf8` but just on different versions.

See also:
- https://dev.mysql.com/worklog/task/?id=10778
- https://dev.mysql.com/worklog/task/?id=10353
- https://mariadb.com/kb/en/unicode/